### ISSUE TITLE
Components: plugin or extraction?

### ISSUE THREAD

**Issue by missmatsuko** [tailwindcss/discuss#197]
I see there's 2 ways to create components in Tailwind:
1. [Create a component plugin (JS)](https://tailwindcss.com/docs/plugins/#adding-components)
2. [Extract components (CSS, kind of)](https://tailwindcss.com/docs/extracting-components/)

When would you choose one method over another?

Here's a couple thoughts I had:
1. Create a component plugin (JS)
- CSS-in-JS syntax
- Since we are in JS, it's easier to switch between config and component creation
- [Can expose custom options](https://tailwindcss.com/docs/plugins/#exposing-options)
- Easier to re-use between projects?

2. Extract components (CSS, kind of)
- CSS-like syntax
- Since we are applying existing classes, it's easier to switch between markup and component creation
- Unsure if components created with this method can be applied with the `@apply` directive?

**Comment by benface** [comment]
I think your thoughts are pretty spot on, especially that component plugins are easier to re-use between projects and can be customized with options. So I would create a plugin if it's something that is generic / not specific to the project, and use @apply otherwise.

**Comment by benface** [comment]
> Unsure if components created with this method can be applied with the @apply directive?

Not really. `@apply` is really meant for utility classes as far as I understand it.
